src/main/java/minicraft/core/Renderer.java

```diff
@@ -114,16 +114,28 @@ public static void initScreen() {
 	public static void render() {
 		if (screen == null) return; // No point in this if there's no gui... :P
 
-		if (readyToRenderGameplay) {
-			renderLevel();
-			if (player.renderGUI) renderGui();
-		}
+		if (Game.reloading) {
+			screen.render(0, 0, Screen.w, Screen.h, 0xFF_FFFFFF);
 
-		if (display != null) // Renders menu, if present.
-			display.render(screen);
+			if (Game.currentLoader != null) {
+				int barw = 120;
+				int barh = 8;
 
-		if (!canvas.hasFocus()) renderFocusNagger(); // Calls the renderFocusNagger() method, which creates the "Click to Focus" message.
+				screen.render((Screen.w - barw) / 2, (Screen.w - barh) / 2, barw, barh, 0xFF_FFFFFF);
+				screen.render((Screen.w - barw) / 2 + 1, (Screen.w - barh) / 2 + 1, barw - 2, barh - 2, 0xFF_000000);
+				screen.render((Screen.w - barw) / 2 + 2, (Screen.w - barh) / 2 + 2, (int)(barw * Game.currentLoader.getProgress()) - 4, barh - 4, 0xFF_FFFFFF);
+			}
+		} else {
+			if (readyToRenderGameplay) {
+				renderLevel();
+				if (player.renderGUI) renderGui();
+			}
 
+			if (display != null) // Renders menu, if present.
+				display.render(screen);
+
+			if (!canvas.hasFocus()) renderFocusNagger(); // Calls the renderFocusNagger() method, which creates the "Click to Focus" message.
+		}
```

Comment:
I think this should really be handled in another display class, also referencing my previous comment to the reload field.